src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx

```diff
@@ -60,19 +60,11 @@ function NewContactMethodPage({route}: NewContactMethodPageProps) {
     const addNewContactMethod = useCallback(
         (magicCode: string) => {
             User.addNewContactMethod(addSMSDomainIfPhoneNumber(pendingContactAction?.contactMethod ?? ''), magicCode);
+            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHOD_DETAILS.getRoute(pendingContactAction?.contactMethod ?? ''));
```

Comment:
With this change, even when the user submits an incorrect magic code while validating the primary contact, it navigates to the secondary contact method page. I think the validation code form for the primary contact should remain visible until the backend confirms the code entered is correct. If the backend indicates the code is incorrect, an error message should be displayed on the primary contact form itself. Only after the backend confirms the code’s validity should the user be navigated to the validation code form for the newly added contact method.


https://github.com/user-attachments/assets/f95167df-50bd-4dee-8e84-11fb341dbd33

Reply:
I think there was a check for that, which I removed. I'll add it back